qutebrowser/mainwindow/tabbedbrowser.py

```diff
@@ -126,6 +127,7 @@ def __init__(self, *, win_id, private, parent=None):
         self.cur_fullscreen_requested.connect(self.widget.tabBar().maybe_hide)
         self.widget.setSizePolicy(QSizePolicy.Expanding, QSizePolicy.Expanding)
         self._undo_stack = []
```

Comment:
This is quite confusing because it doesn't get used after all - why not remove it?

Reply:
The problem is that `_update_stack_size` only *resize* the undo stack, so it must be an iterable before the call.

Reply:
Ah, right, I missed that discussion - my bad! I agree the changing type is a bit confusing, IMHO this would be better:

```suggestion
        self._undo_stack = collections.deque()
```

(or alternatively, `self._undo_stack = None` and adjust `_update_stack_size` to check for that - but setting it to an empty deque is simpler IMHO).

As for handling conflicts: I'd prefer to look at each PR in isolation where possible, and deal with conflicts as they arise. If anything of that breaks https://gitlab.com/jgkamat/dotfiles/blob/master/qutebrowser/.config/gen_integration.sh I'd be happy to merge master into the offending PR-branch and fix the conflicts, shouldn't be a big deal. IMHO that's the easier way to deal with the current situation, and it avoids not fixing/changing things because conflicts could arise. @jgkamat what do you think?

Reply:
> If anything of that breaks gen_integration.sh I'd be happy to merge master into the offending PR-branch and fix the conflicts, shouldn't be a big deal.

I don't really understand this. If it's "fixed" for an integration branch, then there aren't any conflicts with master by definition. As far as I know, the only way of publicizing merge conflicts is by doing the merge. I could stack pulls on-top of old ones where the conflicts are really blocking me (I'm already doing this in some places). In this case, it dosen't matter, as the offending PR got merged.

Reply:
> As far as I know, the only way of publicizing merge conflicts is by doing the merge.

Let's say PR 1 (this one) lands in master, thus you can't merge PR 2 cleanly anymore. There are (at least) three ways to fix and publicize the conflict resolution:

- Merge PR 2 to master.
- Rebase PR 2 on the latest master and force-push it. Nice for WIP stuff nobody reviewed yet, but other than that, usually not the right thing to do IMHO.
- Merge master into PR 2. That way, the PR branch has all the latest changes from master, and also cleanly merges to master again.

Here I was talking about the last approach.

Reply:
If one of the two PRs is already on master, there's no issue at all (any of the options above would work). The issue is only present when both A and B aren't on master, but A and B are whitelisted as "good to merge". Hence, when #4723 was merged, there were no more issues with changing this line.

Reply:
Fair point, I don't have a good solution to that - other than you telling me where the pain points are, so I can redirect my attention there. Either way, hopefully the PR count goes down rather than up now!